src/main/java/ca/corefacility/bioinformatics/irida/ria/web/PasswordResetAjaxController.java

```diff
@@ -46,11 +48,66 @@ public PasswordResetAjaxController(UIPasswordResetService UIPasswordResetService
 	public ResponseEntity<AjaxResponse> adminNewPasswordReset(@PathVariable Long userId, Principal principal,
 			Locale locale) {
 		try {
-			return ResponseEntity.ok(
-					new AjaxSuccessResponse(UIPasswordResetService.adminNewPasswordReset(userId, principal, locale)));
+			return ResponseEntity.ok(new AjaxSuccessResponse(service.adminNewPasswordReset(userId, principal, locale)));
 		} catch (UIEmailSendException e) {
 			return ResponseEntity.status(HttpStatus.CONFLICT).body(new AjaxErrorResponse(e.getMessage()));
 		}
 	}
 
+	/**
+	 * Create a password reset for the given email address or username
+	 *
+	 * @param usernameOrEmail The email address or username to create a password reset for
+	 * @param locale          The logged in user's locale
+	 * @return message indicating if the password reset was successfully created or not
+	 */
+	@RequestMapping(method = RequestMethod.POST)
+	public ResponseEntity<AjaxResponse> createAndSendNewPasswordResetEmail(@RequestParam String usernameOrEmail,
+			Locale locale) {
+		try {
+			return ResponseEntity.ok(
+					new AjaxSuccessResponse(service.createAndSendNewPasswordResetEmail(usernameOrEmail, locale)));
+		} catch (EntityNotFoundException e) {
+			return ResponseEntity.ok(new AjaxSuccessResponse(e.getMessage()));
+		} catch (UIEmailSendException e) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(new AjaxErrorResponse(e.getMessage()));
+		}
+	}
+
+	/**
+	 * Activate the user account
+	 *
+	 * @param identifier The ID of the {@link PasswordReset}
+	 * @return response with {@link UserPasswordResetDetails}
+	 */
+	@RequestMapping(value = "/activate_account", method = RequestMethod.POST)
+	public ResponseEntity<UserPasswordResetDetails> activateAccount(@RequestParam String identifier) {
+		try {
+			return ResponseEntity.ok(service.activateAccount(identifier));
+		} catch (UIEmailSendException e) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+		}
+	}
+
+	/**
+	 * Update the password for the {@link User}
+	 *
+	 * @param resetId  The {@link PasswordReset} identifier
+	 * @param password The new password to set for the user
+	 * @param model    A model for the page
+	 * @param locale   The logged in user's locale
+	 * @return response with {@link PasswordResetResponse}
+	 */
+	@RequestMapping(value = "/update_password", method = RequestMethod.POST)
+	public ResponseEntity<PasswordResetResponse> updatePassword(@RequestParam String resetId,
```

Comment:
See comment on PasswordResetResponse

Reply:
Removed in 66fd23c47ca7b634ae1918ac77341a2fdf53048e